stackslib/src/main.rs

```diff
@@ -1166,6 +1191,162 @@ simulating a miner.
         println!("{:?}", inv);
     }
 
+    if argv[1] == "get-nakamoto-tip" {
```

Comment:
Can you move these code blocks into another module, like `src/shadow_blocks.rs` or something?

Reply:
Sure. I can add unit tests that way as well.